compiler/qsc_frontend/src/compile.rs

```diff
@@ -88,8 +90,26 @@ impl SourceMap {
             offset_sources.push(source);
         }
 
+        // Each source has a name, which is a string. The project root dir is calculated as the
+        // common prefix of all of the sources.
+        // Calculate the common prefix.
+        let project_root_dir: String = longest_common_prefix(
```

Comment:
This isn't necessarily true - for example, all the source files could live under `project_root/src/Microsoft/Quantum/`. Best to pass in the project root explicitly from whatever point we're discovering the `qsharp.json`